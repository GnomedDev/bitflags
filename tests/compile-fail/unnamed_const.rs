use bitflags::bitflags;

bitflags! {
    pub struct Unnamed: u8 {
        const _ = 1;

        const A = Self::_.bits();
    }
}

fn main() {}
