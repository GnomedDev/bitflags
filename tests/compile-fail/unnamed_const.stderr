error: expected identifier, found reserved identifier `_`
 --> tests/compile-fail/unnamed_const.rs:7:25
  |
7 |         const A = Self::_.bits();
  |                         ^ expected identifier, found reserved identifier

error[E0599]: no associated item named `_` found for struct `Unnamed` in the current scope
 --> tests/compile-fail/unnamed_const.rs:7:25
  |
3 | / bitflags! {
4 | |     pub struct Unnamed: u8 {
5 | |         const _ = 1;
6 | |
7 | |         const A = Self::_.bits();
  | |                         ^
  | |                         |
  | |                         associated item not found in `Unnamed`
  | |                         help: there is an associated constant with a similar name: `A`
8 | |     }
9 | | }
  | |_- associated item `_` not found for this struct
